lib/gollum-lib/file.rb

```diff
@@ -26,6 +26,13 @@ def url_path
       path
     end
 
+    # Public: The SHA hash identifying this file
+    #
+    # Returns the String SHA.
+    def sha
+      @blob && @blob.id
```

Comment:
It can be `nil` because `File`s are first initialized, and then populated in a second step. This strikes me as an ugly and unnecessary pattern, which we can hopefully change when we rework the `File` and `Page` classes.